File: src/main/java/cope/nebula/client/feature/command/impl/Friend.java
package cope.nebula.client.feature.command.impl;

import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import com.mojang.brigadier.builder.RequiredArgumentBuilder;
import com.mojang.realmsclient.gui.ChatFormatting;
import cope.nebula.asm.duck.IEntityPlayer;
import cope.nebula.client.feature.command.Command;
import cope.nebula.client.feature.command.type.PlayerArgumentType;
import net.minecraft.entity.player.EntityPlayer;

public class Friend extends Command {
    public Friend() {
        super("friend", LiteralArgumentBuilder.literal("friend")
                .then(RequiredArgumentBuilder.argument("player", PlayerArgumentType.player())
                        .executes((ctx) -> {
                            EntityPlayer player = PlayerArgumentType.getPlayer(ctx, "player");
                            if (player == null) {
                                send("Could not find that player.");
                                return 0;
                            }

                            boolean result = ((IEntityPlayer) player).friend();
                            send(result ? "Added" : "Removed" + " " + ChatFormatting.DARK_PURPLE + player.getName() + ChatFormatting.RESET + " as a friend.");

                            return 0;
                        }))
                .executes((ctx) -> {
                    send("Please provide a player username/uuid");
                    return 0;
                }));
    }
}

File: src/main/kotlin/wtf/nebula/client/feature/command/impl/Friend.kt
package wtf.nebula.client.feature.command.impl

import com.mojang.brigadier.Command.SINGLE_SUCCESS
import com.mojang.brigadier.arguments.StringArgumentType
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import com.mojang.brigadier.builder.RequiredArgumentBuilder
import wtf.nebula.client.Nebula
import wtf.nebula.client.feature.command.Command
import wtf.nebula.client.registry.impl.CommandRegistry.Companion.INVALID_USAGE

class Friend : Command(arrayOf("friend", "fren", "f"), "[player name]", "Friends or unfriends a player") {
    override fun build(builder: LiteralArgumentBuilder<Any>) {
        builder.then(RequiredArgumentBuilder.argument<Any?, String?>("friendName", StringArgumentType.word())
            .executes {
                val name = StringArgumentType.getString(it, "friendName")
                    ?: return@executes INVALID_USAGE

                if (Nebula.friendManager.isFriend(name)) {
                    Nebula.friendManager.remove(name)
                    send("Removed that player as a friend")
                }

                else {
                    Nebula.friendManager.add(name)
                    send("Added that player as a friend")
                }

                return@executes SINGLE_SUCCESS
            })
            .executes {
                return@executes INVALID_USAGE
            }
    }
}